### ISSUE TITLE
Dynamic query support

### ISSUE THREAD

**Issue by Xuanwo** [singularity-data/sqllogictest-rs#13]
Hi, is there a plan to support dynamive query like the following?

```shell
SELECT now();
```

**Comment by skyzh** [comment]
Queries should be supported by database systems. sqllogictest-rs simply parses the slt file and sends the sql to the corresponding database systems.

**Comment by Xuanwo** [comment]
Oh, let me correct the original question: will sqllogictest has native support for testing dynamic query?

## Background

Databend has system tables which contains current system time, we test it via the following logic:

- A SQL file

```sql
SELECT * from system.tables where name = 'tables';
```

- A result file (just like `sqllogictest-rs` does, but in different file)

```text
system	tables	SystemTables	yyyy-mm-dd HH:MM:SS.sss +0000
```

- A result filter (we can't use the result directly, we need to do some filter on it)

```text
\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d[.]\d\d\d [+-]\d\d\d\d
yyyy-mm-dd HH:MM:SS.sss +0000
```

We use `sed` and related to tools to do this hacky job.

## Sqllogictest

Let's come back to `sqllogictest-rs`.

Maybe we can natively support something like pattern match?

**Comment by skyzh** [comment]
Get it! I think it's okay to support custom logic (like regex) in sqllogictest. But there are more things to consider:

* Query result generally have multiple rows and multiple columns. What's the grammar for such cases?
* Where to put the regex in `.slt` file?

**Comment by Xuanwo** [comment]
Great, I will start a proposal for this feature!

**Comment by Xuanwo** [comment]
Implemented in #15

**Comment by xudong963** [comment]
Though we support `Validator`, but it seems still a little difficult to support https://github.com/risinglightdb/sqllogictest-rs/issues/13#issuecomment-1016088811
